broker/broker.go

```diff
@@ -0,0 +1,87 @@
+//  This file is part of the eliona project.
+//  Copyright © 2022 LEICOM iTEC AG. All Rights Reserved.
+//  ______ _ _
+// |  ____| (_)
+// | |__  | |_  ___  _ __   __ _
+// |  __| | | |/ _ \| '_ \ / _` |
+// | |____| | | (_) | | | | (_| |
+// |______|_|_|\___/|_| |_|\__,_|
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+//  BUT NOT LIMITED  TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NON INFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+//  DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+package broker
+
+import (
+	"fmt"
+	"reflect"
+	"template/apiserver"
+
+	"github.com/eliona-smart-building-assistant/go-utils/common"
+)
+
+type ExampleDevice struct {
+	ID   string
+	Name string
+}
+
+func GetTags(config apiserver.Configuration) ([]ExampleDevice, error) {
+	return nil, nil
+}
+
+func (tag *ExampleDevice) AdheresToFilter(filter [][]apiserver.FilterRule) (bool, error) {
+	f := apiFilterToCommonFilter(filter)
+	fp, err := structToMap(tag)
+	if err != nil {
+		return false, fmt.Errorf("converting strict to map: %v", err)
+	}
+	adheres, err := common.Filter(f, fp)
+	if err != nil {
+		return false, err
+	}
+	return adheres, nil
+}
+
+func structToMap(input interface{}) (map[string]string, error) {
```

Comment:
Good idea to put all struct fields into a map :-)

Reply:
For the future we can rewrite this methos and add fields recursive from structs in structs.

Reply:
Definitely, I forgot about nested structs.